src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -1144,4 +1045,213 @@ trait BronzeTransforms extends SparkSessionWrapper {
+    try {
+      if(auditLogConfig.sqlEndpoint.getOrElse("").nonEmpty) {
+        val host = apiEnv.workspaceURL.stripPrefix("https://").stripSuffix("/")
+        val sqlEndpoint = auditLogConfig.sqlEndpoint.get
+        val query =
+          s"""select * from ${auditLogConfig.systemTableName.get.toString}
+             |where workspace_id='${organizationId}'
+             |and event_time >= '${fromTimeSysTableCompatible}'
+             |and event_time <= '${untilTimeSysTableCompatible}'
+             |""".stripMargin
+        logger.log(Level.INFO, query)
+        val systemTableNameDf = spark.read
+          .format("databricks")
+          .option("host", host)
+          .option("httpPath", sqlEndpoint)
+          .option("personalAccessToken", apiEnv.rawToken)
+          .option("query", query)
+          .load()
+        systemTableNameDf
+      }
+      else{
+        spark.table(auditLogConfig.systemTableName.get.toString)
+          .filter('workspace_id === organizationId)
+          .filter('event_time >= fromTimeSysTableCompatible
+            && 'event_time <= untilTimeSysTableCompatible)
+      }
+    } catch {
+      case e: org.apache.spark.sql.AnalysisException =>
+        throw new Exception(s"Access issue with table system.access.audit: ${e.getMessage}")
+    }
+  }
+
+
+  def getAuditLogsDfFromSystemTables(
+                               fromTime: LocalDateTime,
+                               untilTime: LocalDateTime,
+                               organizationId: String,
+                               auditLogConfig: AuditLogConfig,
+                               apiEnv: ApiEnv
+                             ): DataFrame = {
+    try {
+      println(s"Fetching data from system.access.audit for workspace_id - ${organizationId}")
+      val sysTableFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS")
+      // Adding the below code to add time so that the whole day data can be fetched, despite of the fromTime and untilTime
+      val fromTimeSysTableCompatible = fromTime.withHour(0).withMinute(0).withSecond(0).format(sysTableFormat)
+      val untilTimeSysTableCompatible = untilTime.withHour(23).withMinute(59).withSecond(59).format(sysTableFormat)
+      println(s"system.access.audit fromTime - ${fromTimeSysTableCompatible}")
+      println(s"system.access.audit untilTime - ${untilTimeSysTableCompatible}")
+
+      val rawSystemTableFiltered = fetchDatafromSystemTableAuditLog(fromTimeSysTableCompatible,
+                                                                    untilTimeSysTableCompatible,
+                                                                    organizationId,
+                                                                    auditLogConfig,
+                                                                    apiEnv
+                                                                    )
+
+      if (rawSystemTableFiltered.isEmpty) {
```

Comment:
Will this end up marking the module as empty or stopping the job entirely?

Reply:
this will stop the job since this is a failure in audit_log